client/handlers.go

```diff
@@ -0,0 +1,78 @@
+package client
+
+import (
+	"github.com/datastax/go-cassandra-native-protocol/frame"
+	"github.com/datastax/go-cassandra-native-protocol/message"
+	"github.com/rs/zerolog/log"
+	"strings"
+)
+
+// A RequestHandler to handle server-side heartbeats. This handler assumes that every OPTIONS request is a heartbeat
+// probe and replies with a SUPPORTED response.
+var HeartbeatHandler RequestHandler = func(request *frame.Frame, conn *CqlServerConnection, _ RequestHandlerContext) (response *frame.Frame) {
+	if _, ok := request.Body.Message.(*message.Options); ok {
+		log.Debug().Msgf("%v: [heartbeat handler]: received heartbeat probe", conn)
+		response, _ = frame.NewResponseFrame(request.Header.Version, request.Header.StreamId, nil, nil, nil, &message.Supported{}, false)
+	}
+	return
+}
+
+// A RequestHandler to handle USE queries. This handler intercepts QUERY requests with a USE statement and replies
+// with a message.SetKeyspaceResult. The provided callback function will be invoked with the new keyspace.
+func NewSetKeyspaceHandler(onKeyspaceSet func(string)) RequestHandler {
+	return func(request *frame.Frame, conn *CqlServerConnection, _ RequestHandlerContext) (response *frame.Frame) {
+		if query, ok := request.Body.Message.(*message.Query); ok {
+			q := strings.TrimSpace(strings.ToLower(query.Query))
+			q = strings.Join(strings.Fields(q), " ")
+			if strings.HasPrefix(q, "use ") {
+				keyspace := strings.TrimPrefix(q, "use ")
+				onKeyspaceSet(keyspace)
+				log.Debug().Msgf("%v: [set keyspace handler]: received USE %v", conn, keyspace)
+				response, _ = frame.NewResponseFrame(request.Header.Version, request.Header.StreamId, nil, nil, nil, &message.SetKeyspaceResult{Keyspace: keyspace}, false)
+			}
+		}
+		return
+	}
+}
+
+// A RequestHandler to handle USE requests. This handler intercepts REGISTER requests and replies with READY.
+var RegisterHandler RequestHandler = func(request *frame.Frame, conn *CqlServerConnection, _ RequestHandlerContext) (response *frame.Frame) {
```

Comment:
is it necessary to assign a variable?

Reply:
My intent was to make it clear that this is a `RegisterHandler` and should be used as such, iow as a value – not as a function that you would call yourself.

Reply:
Another option is to have a function that returns a `RegisterHandler` but it seems kind of pointless

Reply:
Yeah I used that option in a few places, but only when I need to create a closure. Otherwise it's a bit pointless, I agree.